src/Sniffer.php

```diff
@@ -0,0 +1,281 @@
+            $config->generator = null;
+            $config->explain = false;
+            $config->interactive = false;
+            $config->cache = false;
+            $config->showSources = false;
+            $config->recordErrors = false;
+            $config->reportFile = null;
+            $config->reports = [FixerReport::class => null];
+            $config->dieOnUnknownArg = false;
+        }
+
+        return $config;
+    }
+
+    /**
+     * Finds all fenced codeblocks for the relevant language in all the markdown files
+     */
+    private function findFencedCodeblocks(FileList $paths, string $lintLanguage): array
+    {
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            echo 'Finding fenced codeblocks... ';
+        }
+
+        $blocks = [];
+
+        /** @var string $path */
+        foreach ($paths as $path => $v) {
+            $document = $this->parser->parse(file_get_contents($path));
+            $codeBlocks = $this->query->findAll($document);
+
+            $n = 0;
+            /** @var FencedCode $block */
+            foreach ($codeBlocks as $block) {
+                if (strtoupper($block->getInfo()) !== $lintLanguage) {
+                    continue;
+                }
+                // We only want to count relevant code blocks
+                $n++;
+
+                // $startAt is the line in the md file where the ```php line sits
+                $startAt = $block->getStartLine();
+
+                // Pad the content out so we have an accurate line count, and prepend a php code opening tag
+                $content = str_repeat(PHP_EOL, $startAt - 1);
+                $content .= '<?php' . PHP_EOL . $block->getLiteral();
+
+                // Report each block separately (by making the path unique) so it's treated as its own file
+                // This lets us lint for things like namespaces more easily since the namespace in an earlier block
+                // won't be counted towards a later block in the same file
+                $key = dirname($path) . '/' . basename($path, '.md') . "_{$n}" . '.md';
+                $blocks[$key] = [
+                    'content' => $content,
+                    'path' => $key,
+                    'realpath' => $path,
+                    'num' => $n,
+                    'md' => $block,
+                ];
+            }
+        }
+
+        if (PHP_CODESNIFFER_VERBOSITY > 0) {
+            $numBlocks = count($blocks);
+            echo "DONE ($numBlocks codeblocks in queue)" . PHP_EOL;
+        }
+
+        return $blocks;
+    }
+
+    /**
+     * Run the codesniffing rules over the identified markdown codeblocks
+     *
+     * This is very nearly a direct copy of Runner::run()
+     */
+    private function sniff(Runner $sniffer, array $todo): int
+    {
+        // Turn all sniff errors into exceptions.
+        set_error_handler([$sniffer, 'handleErrors']);
+
+        $lastDir  = '';
```

Comment:
```suggestion
        $lastDir = '';
```

Reply:
Oops I should add a linting config so this gets linted in CI

Reply:
Done